framework/PostProcessors/FTgate.py

```diff
@@ -0,0 +1,293 @@
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  if len(value)>1:
+    raise IOError('NOT gate has received in input ' + str(len(value)) + ' values instead of 1.')
+  if value[0]==0:
+    outcome = 1
+  else:
+    outcome = 0
+  return outcome
+
+def ANDgate(argumentValues):
+  """
+    Method that evaluates the AND gate
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  if 0 in argumentValues:
+    outcome = 0
+  else:
+    outcome = 1
+  return outcome
+
+def ORgate(argumentValues):
+  """
+    Method that evaluates the OR gate
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  if 1 in argumentValues:
+      outcome = 1
+  else:
+      outcome = 0
+  return outcome
+
+def NANDgate(argumentValues):
+  """
+    Method that evaluates the NAND gate
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  out = []
+  out.append(ANDgate(argumentValues))
+  outcome = NOTgate(out)
+  return outcome
+
+def NORgate(argumentValues):
+  """
+    Method that evaluates the NOR gate
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  out = []
+  out.append(ORgate(argumentValues))
+  outcome = NOTgate(out)
+  return outcome
+
+def XORgate(argumentValues):
+  """
+    Method that evaluates the XOR gate
+    The XOR gate gives a true (1 or HIGH) output when the number of true inputs is odd.
+    https://electronics.stackexchange.com/questions/93713/how-is-an-xor-with-more-than-2-inputs-supposed-to-work
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  if argumentValues.count(1.) % 2 != 0:
+    outcome = 1
+  else:
+    outcome = 0
+  return outcome
+
+def IFFgate(argumentValues):
+  """
+    Method that evaluates the IFF gate
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  out = []
+  out.append(XORgate(argumentValues))
+  outcome = NOTgate(out)
+  return outcome
+
+def IMPLYgate(argumentValues):
+  """
+    Method that evaluates the IMPLY gate
+    Note that this gate requires a specific definition of the two inputs. This definition is specifed in the order of the events provided in the input file
+    As an example, BE1->BE2 is translated as:
+      <define-gate name="TOP">
+          <imply>
+              <basic-event name="BE1"/>
+              <basic-event name="BE2"/>
+          </imply>
+      </define-gate>
+    @ In, argumentValues, list, list of values
+    @ Out, outcome, float, calculated outcome of the gate
+  """
+  keys = argumentValues.keys()
+  if argumentValues[keys[0]]==1 and argumentValues[keys[1]]==0:
+    outcome = 0
+  else:
+    outcome = 1
+  return outcome
+
+def ATLEASTgate(argumentValues,k):
```

Comment:
not camel back...valid for all the other "gate" evaluators

Reply:
fixed.